packages/charts/react-charting/src/components/DeclarativeChart/PlotlySchemaAdapter.ts

```diff
@@ -402,6 +403,29 @@ export const transformPlotlyJsonToVSBCProps = (
   };
 };
 
+const getColorFromScale = (value: number, scale: Array<[number, string]>, domain: [number, number]): string => {
+  const [dMin, dMax] = domain;
+  if (dMax === dMin) {
+    return scale[0][1];
+  }
+  const norm = (value - dMin) / (dMax - dMin);
+  // Find two colorscale stops surrounding norm
+  for (let i = 1; i < scale.length; i++) {
+    if (norm <= scale[i][0]) {
+      const [leftPos, leftColor] = scale[i - 1];
+      const [rightPos, rightColor] = scale[i];
+      // Linear interpolate between the two colors using d3-scale and d3-interpolate
+      const colorInterpolator = d3ScaleLinear<string>()
+        .domain([leftPos, rightPos])
+        .range([leftColor, rightColor])
+        .interpolate(interpolateRgb);
+
+      return colorInterpolator(norm);
+    }
+  }
+  return scale[scale.length - 1][1];
```

Comment:
we already have an implementation of using colorscale and color axis.

See line 897-930 in this file